app/src/main/java/com/github/muellerma/prepaidbalance/ui/MainActivity.kt

```diff
@@ -203,9 +211,12 @@ class MainActivity : AbstractBaseActivity(), SwipeRefreshLayout.OnRefreshListene
                 CheckResult.USSD_FAILED -> {
                     showSnackbar(R.string.ussd_failed)
                 }
-                CheckResult.MISSING_PERMISSIONS -> {
+                CheckResult.MISSING_PERMISSIONS_PHONE -> {
                     requestPermissionLauncher.launch(arrayOf(CALL_PHONE, READ_PHONE_STATE))
                 }
+                CheckResult.MISSING_PERMISSIONS_NOTIFICATION -> {
+                    requestPermissionLauncher.launch(arrayOf(POST_NOTIFICATIONS))
```

Comment:
Better check this in `onResume()` and open a `MaterialAlertDialog` that asks to grant permissions?

Reply:
Sounds good.. Will try that